mlflow/entities/metric.py

```diff
@@ -63,6 +69,11 @@ def model_id(self):
         """ID of the Model associated with the metric."""
         return self._model_id
 
+    @property
+    def dataset_uuid(self) -> Optional[str]:
+        """String. UUID of the dataset associated with the metric."""
+        return self._dataset_uuid
+
```

Comment:
This is an internal ID (for now) and should not be exposed on the metric entity.

Reply:
Gotcha, will remove it.